network/crl.go

```diff
@@ -127,6 +128,66 @@ func (e CRLError) Unwrap() error {
 	return e.err
 }
 
+// NewCRLConfigByName return initialized CRLConfig config using flags registered with RegisterCertVerifierArgsForService
+func NewCRLConfigByName(flags *flag.FlagSet, name string, namerFunc NamerFunc) (*CRLConfig, error) {
+	var crlURL, crlFromCert string
+	var crlCheckOnlyLeafCertificate bool
+	var crlCacheSize, crlCacheTime uint
+	if f := flags.Lookup(namerFunc(name, "url", "crl")); f != nil {
+		crlURL = f.Value.String()
+		if crlURL == "" {
+			crlURL = tlsCrlURL
+		}
+	}
+	if f := flags.Lookup(namerFunc(name, "from_cert", "crl")); f != nil {
+		crlFromCert = f.Value.String()
+		if crlFromCert == "" {
+			crlFromCert = tlsCrlFromCert
+		}
+	}
+	if f := flags.Lookup(namerFunc(name, "check_only_leaf_certificate", "crl")); f != nil {
+		getter, ok := f.Value.(flag.Getter)
+		if !ok {
+			log.Fatal("Can't cast flag's Value to Getter")
+		}
+		val, ok := getter.Get().(*bool)
+		if val == nil {
+			crlCheckOnlyLeafCertificate = tlsCrlCheckOnlyLeafCertificate
+		} else {
+			if !ok {
+				log.WithField("value", getter.Get()).Fatalf("Can't cast %s to boolean value",
+					namerFunc(name, "check_only_leaf_certificate", "crl"))
+			}
+			crlCheckOnlyLeafCertificate = *val
+		}
+	}
+	if f := flags.Lookup(namerFunc(name, "cache_size", "crl")); f != nil {
+		getter, ok := f.Value.(flag.Getter)
+		if !ok {
+			log.Fatal("Can't cast flag's Value to Getter")
+		}
+		val, ok := getter.Get().(uint)
+		if !ok {
+			log.WithField("value", getter.Get()).Fatalf("Can't cast %s to integer value",
+				namerFunc(name, "cache_size", "crl"))
+		}
+		crlCacheSize = val
+	}
+	if f := flags.Lookup(namerFunc(name, "cache_time", "crl")); f != nil {
+		getter, ok := f.Value.(flag.Getter)
```

Comment:
Don't you think to use just some `Parse` methods for flag parsing?

```
crlCacheSize, err = strconv.ParseUint(f.Value.String(), 10, 0)
if err != nil {
      log.WithField("value", getter.Get()).Fatalf("Can't cast %s to integer value", namerFunc(name, "cache_time", "crl"))
}
```

It will reduce the amount of code plus we can avoid these runtime interface castings.

Reply:
you are right. will do